scalding-core/src/main/scala/com/twitter/scalding/graph/Neighbors.scala

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding.graph
+
+sealed trait Neighbors[T] {
+  def neighbors: Array[T]
```

Comment:
Don't need mutation but what is really important here is the least amount of memory overhead for the Neighbors.  Moving from Set -> Array saves a lot of memory.  Does Vector have memory characteristics close to native arrays?

Also about the classtags above, Array needs a classtag so it can be constructed.